extensions/wikia/ArticleVideo/scripts/ooyala-player.js

```diff
@@ -0,0 +1,120 @@
+
+	function OoyalaHTML5Player(container, params, onPlayerCreate) {
+		var playerWidth = container.scrollWidth;
+
+		this.params = params;
+		this.params.width = playerWidth;
+		this.params.height = Math.floor((playerWidth * 9) / 16);
+		this.params.onCreate = this.onCreate.bind(this);
+
+		this.onPlayerCreated = onPlayerCreate;
+
+		this.params.skin = {
+			config: baseJSONSkinUrl
+		};
+
+		this.containerId = container.id;
+		this.player = null;
+	}
+
+	OoyalaHTML5Player.prototype.setUpPlayer = function () {
+		this.createPlayer();
+	};
+
+	/**
+	 * @returns {void}
+	 */
+	OoyalaHTML5Player.prototype.createPlayer = function () {
+		var self = this;
+		window.OO.ready(function () {
+			self.player = OO.Player.create(self.containerId, self.params.videoId, self.params);
+		});
+	};
+
+	/**
+	 * @param {*} player
+	 * @returns {void}
+	 */
+	OoyalaHTML5Player.prototype.onCreate = function (player) {
+		var messageBus = player.mb,
+			self = this;
+
+		this.onPlayerCreated(player);
+
+		messageBus.subscribe(window.OO.EVENTS.PLAYBACK_READY, 'ui-update', function () {
+			self.onPlaybackReady()
+		});
+	};
+
+	/**
+	 * Called when the video is at the "start" state screen
+	 *
+	 * @returns {void}
+	 */
+	OoyalaHTML5Player.prototype.onPlaybackReady = function () {
+		var durationElem = document.createElement('div'),
+			playButton = document.getElementById(this.containerId).querySelector('.oo-start-screen .oo-action-icon'),
+			screenInfo = document.getElementById(this.containerId).querySelector('.oo-state-screen-info');
+
+		// Add video duration to start screen
+		durationElem.innerHTML = mw.message('articlevideo-watch', this.getFormattedDuration(this.player.getDuration())).text().toUpperCase();
+		durationElem.classList.add('video-duration');
+		screenInfo.insertBefore(durationElem, screenInfo.firstChild);
+
+		// Replace default play icon with ours.
+		playButton.innerHTML = playIcon;
+	};
+
+	/**
+	 * Formats milliseconds as HH:mm:ss duration
+	 *
+	 * @param {number}
+	 * @returns {string}
+	 */
+	OoyalaHTML5Player.prototype.getFormattedDuration = function (millisec) {
+		var seconds = parseInt(millisec / 1000, 10);
+		var hours = parseInt(seconds / 3600, 10);
+		seconds = seconds % 3600;
+		var minutes = parseInt(seconds / 60, 10);
+		seconds = seconds % 60;
+		var duration = '';
+		if (hours > 0) {
+			duration += hours + ':';
+			if (minutes < 10) {
+				duration += '0';
+			}
+		}
+		duration += minutes + ':';
+		if (seconds < 10) {
+			duration += '0';
+		}
+		return duration + seconds;
+	};
+
+	OoyalaHTML5Player.prototype.hideControls = function () {
+		$('.oo-control-bar').css('visibility', 'hidden');
+	};
+
+	OoyalaHTML5Player.prototype.showControls = function () {
+		$('.oo-control-bar').css('visibility', 'visible');
+	};
+
+	OoyalaHTML5Player.initHTMl5Players = function (videoElementId, playerParams, videoId, onCreate) {
```

Comment:
And why is it called initHTML5Player**s**? It inits and returns a single player.